### ISSUE TITLE
FileWriter is not thread safe

### ISSUE THREAD

**Issue by bingranl** [Kotlin/dokka#2308]
**Describe the bug**
FileWriter doesn't seem to be thread safe. Android Gradle Plugin integrates with dokka-core and its javadoc plugin to generate javadoc for java/kotlin sources. In order to have more parallelism, we try to launch dokka via gradle workers so that there would more than one dokka generation tasks running at the same time. In this case, we notice test failures(flaky) from [FileWriter](https://github.com/Kotlin/dokka/blob/e9f5da45c0fcfec5f7c150229301904d7915e090/plugins/base/src/main/kotlin/renderers/FileWriter.kt#L74) complaining stream is closed when reading bytes.

see stacktrace
```
Caused by: java.io.IOException: Stream closed
	at kotlin.io.ByteStreamsKt.copyTo(IOStreams.kt:110)
	at kotlin.io.ByteStreamsKt.copyTo$default(IOStreams.kt:103)
	at kotlin.io.ByteStreamsKt.readBytes(IOStreams.kt:136)
	at org.jetbrains.dokka.base.renderers.FileWriter$copyFromJar$3.invokeSuspend(FileWriter.kt:74)
	at org.jetbrains.dokka.base.renderers.FileWriter$copyFromJar$3.invoke(FileWriter.kt)
```
I noticed dokka-gradle-plugin doesn't use gradle workers, which is also a sign that dokka-core doesn't is not thread safe.

**Expected behaviour**
FileWriter/Dokka-core to be thread safe. 

**Installation**
- Operating system: Linux
- Build tool: Gradle 7.3.3 Android Gradle plugin 7.2.0-alpha07
- Dokka version: 1.4.30

**Comment by IgnatBeresnev** [comment]
A similar issue was fixed in Dokka 1.6: #2205

Are you experiencing this with newer versions of Dokka? 1.4 is quite old

**Comment by bingranl** [comment]
I could try 1.6 and see how it goes.

**Comment by IgnatBeresnev** [comment]
It's best to try the latest version possible, ideally 1.7.20 or something close. That would be helpful, thanks

**Comment by IgnatBeresnev** [comment]
@bingranl hi! Any update on this?

**Comment by IgnatBeresnev** [comment]
I'll close the issue as resolved as there haven't been any reports related to this for half a year, and we suspect was fixed in #2205.

Please, let us know if you encounter this problem again, and we'll re-open the issue and investigate.
